Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+
+It is recommended to have your extension be `.<something>.json` where the `<something>` is related to your node type (e.g. `.mrk.json` for Markups).
+
+:::
+
+#### The reader
+
+The recommended way to read a file into a MRML node is through the storage node. The reader, on the other hand, exists to interface with the loading facilities of 3D Slicer (drag and drop, as well as the button to load data into the scene). As such, the reader uses the storage node in its implementation.
+
+
+Files:
+
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Reader.h
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Reader.cxx
+
+Key Points:
+
+* Naming convention for class: `qSlicer<MyCustomType>Reader`
+* Inherits from [`qSlicerFileReader`](http://apidocs.slicer.org/master/classqSlicerFileReader.html).
+* In the class definition, the following macros should be used:
+  * `Q_OBJECT`
+  * `Q_DECLARE_PRIVATE`
+  * `Q_DISABLE_COPY`
+* Constructing a new node:
+  * Create constructor `qSlicer<MyCustomType>Reader(QObject* parent = nullptr)`.
+    * This constructor, even if it is not explicitly used, allows this file to be wrapped in Python.
+* Override [`QString description() const`](http://apidocs.slicer.org/master/classqSlicerIO.html#af44106dbf852df0e6cc836b377630f5e) to provide a short description on the types of files read.
+* Override [`IOFileType fileType() const`](http://apidocs.slicer.org/master/classqSlicerIO.html#aac38b570ec5c0692caefa7d87657e58b) to give a string to associate with the types of files read.
+  * This string can be used in conjunction with the python method [`slicer.util.loadNodeFromFile`](https://slicer.readthedocs.io/en/latest/developer_guide/slicer.html#slicer.util.loadNodeFromFile)
+* Override [`QStringList extensions() const`](http://apidocs.slicer.org/master/classqSlicerFileReader.html#afb3187915977b2253d86634cc465b23d) to provide the extensions that can be read.
+  * Should be the same as the storage node since the reader uses the storage node.
+* Override [`bool load(const IOProperties& properties)`](http://apidocs.slicer.org/master/classqSlicerFileReader.html#ac9acb878cd8adcc426e9a7a9edac15df). This is the function that actually loads the node from the file into the scene.
+
+:::{important}
+
+The reader is not a VTK object, like the previous objects discussed. It is actually a QObject, so we follow Qt guidelines. One such guideline is the [D-Pointer pattern](https://wiki.qt.io/D-Pointer), which is recommended for use.
+
+:::
+
+#### The writer
+
+The writer is the companion to the reader, so, similar to the reader, it does not implement the actual writing of files, but rather it uses the storage node. Its existence is necessary to use 3D Slicer’s built in saving facilities, such as the save button.
+
+Files:
+
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Writer.h
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Writer.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- qSlicer<MyCustomType>Writer.h
              |-- qSlicer<MyCustomType>Writer.cxx
```

Reply:
Updated to suggested file structure formatting.